src/components/course/data/tests/hooks.test.jsx

```diff
@@ -1744,7 +1753,9 @@ describe('useCourseListPrice', () => {
       },
       entitlements: [],
     };
-    useCourseMetadata.mockReturnValue({ data: updatedListPrice || getCoursePrice(updatedCourseMetadata) });
+    useCourseMetadata.mockReturnValue(
+      { data: updatedListPrice.length > 0 ? updatedListPrice : getCoursePrice(updatedCourseMetadata) },
+    );
```

Comment:
Is this missing a sub-key under the `data` top-level key? When we call useCourseMetdata, it doesn't normally just plop down the price as a value to the `data` key, does it?

Reply:
In this case it does because of how `useCourseListPrice` utilizing the `select` within the useCourseMetadata arguments. It passes in the data from the transformed `useCourseMetadata` return value to resolve within `getCoursePrice` or the updatedListPrice (acting as a no-op on the original course metadata)

``` jsx
export function useCourseListPrice() {
  const { data: { listPrice } } = useCourseRedemptionEligibility();
  const resolveListPrice = ({ transformed }) => listPrice || getCoursePrice(transformed);
  return useCourseMetadata({
    select: resolveListPrice,
  });
}
```